src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp

```diff
@@ -144,7 +164,13 @@ ConvertMatMulToFullyConnected::ConvertMatMulToFullyConnected() {
         }
 
         // Connect Convert to new input if needed
-        if (is_convert) {
+        if (is_convert && transpose_node && !can_reuse_transpose) {
```

Comment:
Doesn't the second branch lead to the same behavior?

Reply:
The second branch is supposed to be applied in case of FC weights sharing during first FC processing:
Original graph:
```
      Input1 --  FC1
               /          
Weights -- Convert 
               \
      Input2 --  FC2
```
After FC1 processing:
```
                   Input1 --  FC1
                            /          
Weights -- Transpose -- Convert 
                            \
                   Input2 --  FC2
```
If required transpose op for FC2 doesn't match with existing transpose for some reason, we need to add new transpose and corresponding convert node (it's covered by the first branch):
```
                   Input1 --  FC1
                            /          
Weights -- Transpose -- Convert 
        \
          Transpose_new -- Convert_new
                                \
                      Input2 --  FC2
```
Though, not sure if it's possible, but I added this convert processing for consistency